mlos_bench/mlos_bench/services/remote/azure/azure_auth.py

```diff
@@ -47,12 +51,49 @@ def __init__(self,
         # Register methods that we want to expose to the Environment objects.
         self.register([self.get_access_token])
 
+        if "spClientId" in self.config:
+            check_required_params(
+                self.config, {
+                    "keyVaultName",
+                    "certName",
+                    "spClientId",
+                }
+            )
+
         # This parameter can come from command line as strings, so conversion is needed.
         self._req_interval = float(self.config.get("tokenRequestInterval", self._REQ_INTERVAL))
+        keyvault_name = self.config.get("keyVaultName")
```

Comment:
if the key is required, then maybe write
```suggestion
        keyvault_name = self.config["keyVaultName"]
```